cartographer/common/thread_pool.cc

```diff
@@ -46,8 +46,6 @@ ThreadPool::~ThreadPool() {
     MutexLocker locker(&mutex_);
     CHECK(running_);
     running_ = false;
-    CHECK_EQ(task_queue_.size(), 0);
-    CHECK_EQ(tasks_not_ready_.size(), 0);
   }
```

Comment:
This could be a separate (de-flake) PR before adding to the test, but fine.
Please change the commit message to "Deflake destruction and test grpc SerializeState" or the like